PARPACK/SRC/MPI/Makefile.am

```diff
@@ -1,8 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects # Needed as debug/stat* are not in current directory.
 
 F77 = $(MPIF77)
-FFLAGS_SAV = @FFLAGS@
```

Comment:
@sylvestre: OK for you? Was this intentional?

Reply:
Yes this was intentional.  I missed it is still used by `libparpack@LIBSUFFIX@@ITF64SUFFIX@_la_FFLAGS = $(FFLAGS_SAV)` which can now use FFLAGS instead.  Or keep it to a one line change?

Reply:
> libparpack@LIBSUFFIX@@ITF64SUFFIX@_la_FFLAGS = $(FFLAGS_SAV) which can now use FFLAGS instead. 

I guess this is a good move.

> Or keep it to a one line change?

@sylvestre: what do you think? Was this `FFLAGS_SAV` intentional (bug fix?) back in the old days?!

Reply:
I don't know, is there anything interesting in the git blame/log ? :)

Reply:
@turboencabulator:  git-blame is on you! :)
Why did you need `FFLAGS_SAV`? Bug fix? If yes, which one? Can you review this PR?

Reply:
I just moved those lines, they've been there since the beginning.

But I do know why they exist:  Automake sets the default FFLAGS to `-g -O2` which conflicts with the two files we're trying to build without optimizations.  Clearing FFLAGS is a dirty hack to avoid this conflict.  Without it, Makefile looks like this:
```
FFLAGS = -g -O2
...
libparpack_noopt_la_FFLAGS = -O0
...
libparpack_noopt_la-pslamch10.lo: pslamch10.f
        $(AM_V_F77)$(LIBTOOL) $(AM_V_lt) --tag=F77 $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(F77) $(libparpack_noopt_la_FFLAGS) $(FFLAGS) -c -o libparpack_noopt_la-pslamch10.lo `test -f 'pslamch10.f' || echo '$(srcdir)/'`pslamch10.f

libparpack_noopt_la-pdlamch10.lo: pdlamch10.f
        $(AM_V_F77)$(LIBTOOL) $(AM_V_lt) --tag=F77 $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(F77) $(libparpack_noopt_la_FFLAGS) $(FFLAGS) -c -o libparpack_noopt_la-pdlamch10.lo `test -f 'pdlamch10.f' || echo '$(srcdir)/'`pdlamch10.f
```
Since `$(FFLAGS)` comes after `$(libparpack_noopt_la_FFLAGS)`, the `-O2` is last and wins.  So this will break the no-optimization stuff.

Two more questions:
1. What problem was the no-optimization stuff trying to fix?  Do we still need these two files to be built unoptimized?  If not, then let's delete libparpack_noopt.la, build these two files as part of libparpack.la, and delete everything related to FFLAGS in this file.
2. What `FFLAGS` are you trying to add?  Is it something that `configure` should be detecting and adding to `AM_FFLAGS`?

Reply:
I mentioned `AM_FFLAGS` in my previous comment but I just realized that those are getting ignored too.  That might also be part of the problem.

Reply:
> What problem was the no-optimization stuff trying to fix?

Don't know!
@loqs and @turboencabulator: I let you converge to the best solution to PR. Let me know when the PR is ready to be merged.

Reply:
@loqs: is there the same problem with cmake? I guess no.
Otherwise feel free to PR the equivalent fix at cmake side.

Reply:
>     2. What `FFLAGS` are you trying to add?  Is it something that `configure` should be detecting and adding to `AM_FFLAGS`?
`FFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -fstack-clash-protection -fcf-protection  -ffile-prefix-map=/build/arpack/src=/usr/src/debug/arpack`
I detected -fcf-protection was missing from PARPACK.

Reply:
> @loqs: is there the same problem with cmake? I guess no. Otherwise feel free to PR the equivalent fix at cmake side.
autotools only.  cmake is building pdlamch10.f and pslamch10.f with optimization
https://github.com/opencollab/arpack-ng/blob/ba440d13229972b161de37d08dd61bd5a0aa07f2/CMakeLists.txt#L385

Reply:
I was thinking your FFLAGS might be related to MPI, but doesn't look like it.

Usually when I see `-O0` it's because the code has some obscure bug that the optimizer exposes.  But these two files seem fairly simple, and if CMake is building them optimized and nobody has complained yet... I say we remove the no-optimization stuff.